uber/templates/macros.html

```diff
@@ -21,46 +21,205 @@
 {%- endmacro %}
 
 
-{% macro checkbox(model, field_name, id=None, label=None, label_class=None, readonly=False, title=None) -%}
+{% macro checkbox(model, field_name, id=None, label=None, label_class=None, is_readonly=False, title=None) -%}
     {% if label %}
       <label
-        class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}"
-        {% if title %}title="{{ title }}"{% endif %}>
+          class="checkbox-label{% if label_class %} {{ label_class }}{% endif %}{% if is_readonly %} disabled{% endif %}"
+          {% if title %}title="{{ title }}"{% endif %}>
     {% endif %}
-    {% if readonly %}
+    {% if is_readonly %}
       <input
-        type="hidden"
-        name="{{ field_name }}"
-        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
-        value="{%- if model[field_name] -%}1{%- else -%}0{%- endif -%}" />
+          type="hidden"
+          name="{{ field_name }}"
+          id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+          value="{%- if model[field_name] -%}1{%- else -%}0{%- endif -%}" />
       <input
-        type="checkbox"
-        value="1"
-        disabled
-        {% if not label and title %}title="{{ title }}"{% endif %}
-        {% if model[field_name] %}checked{% endif %} />
+          type="checkbox"
+          value="1"
+          disabled
+          {% if not label and title %}title="{{ title }}"{% endif %}
+          {% if model[field_name] %}checked{% endif %} />
     {% else %}
       <input
-        type="checkbox"
-        name="{{ field_name }}"
-        id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
-        value="1"
-        {% if not label and title %}title="{{ title }}"{% endif %}
-        {% if model[field_name] %}checked{% endif %} />
+          type="checkbox"
+          name="{{ field_name }}"
+          id="{% if id %}{{ id }}{% else %}{{ field_name }}{% endif %}"
+          value="1"
+          {% if not label and title %}title="{{ title }}"{% endif %}
+          {% if model[field_name] %}checked{% endif %} />
     {% endif %}
     {% if label %}{{ label }}</label>{% endif %}
 {%- endmacro %}
 
 
-{% macro checkgroup(model, field_name, defaults=False, no_instance=False) -%}
-    {%- set defaults = model[field_name] if not defaults and not no_instance else None -%}
-    {%- set defaults = defaults.split(",") if defaults else [] -%}
+{% macro checkgroup(model, field_name, suffix='', defaults=False, is_readonly=False) -%}
+  {%- set is_class = model is class -%}
+  {%- set defaults = model[field_name] if not defaults and not is_class else '' -%}
```

Comment:
`is_class` will be `True` if `model` is a `class`. There are two ways `checkgroup` can be called:
1. With a model *instance* `checkgroup(attendee, ribbon)`
2. With a model *class* `checkgroup(Attendee, ribbon)`

If you notice, we used to use a parameter called `no_instance` to determine if the given argument was a class or not. I replaced the `no_instance` parameter with a check for `is_class`. Thus `is_class == no_instance`.

If you call `model[field_name]` on an *instance* you get the current model's value, something like `"3456457,1345087"`. If you call `model[field_name]` on a *class* you get the column property, something like `<sqlalchemy.orm.attributes.InstrumentedAttribute object at 0x7fe3fb6a4ca8>`.